onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h

```diff
@@ -101,7 +101,7 @@ struct TensorrtFuncState {
   std::string engine_cache_path;
   nvinfer1::IRuntime* runtime = nullptr;
 
-  nvinfer1::IOptimizationProfile* trt_profile = nullptr;
+  nvinfer1::IOptimizationProfile** trt_profile = nullptr;
```

Comment:
Can you use unique_pointer like engine or context in the structure? Also each engine has its own profile. You may also need to differentiate them in case there are multiple engines created in one model.

Reply:
I think a unique pointer should not be used here. We create the optimization profile through `trt_builder->createOptimizationProfile()` and in its documentation it says 
```C++
    //! The builder retains ownership of the created optimization profile and returns a raw pointer, i.e. the users
    //! must not attempt to delete the returned pointer.
    virtual nvinfer1::IOptimizationProfile* createOptimizationProfile() noexcept = 0;
```
so a unique pointer might not be an option here. 

Concerning the engine, there is only one engine pointer per TensorrtFuncState, and similarly only one trt profile. In my understanding I'm guessing multiple engines will be in separate TensorrtFuncStates (?) so one profile per state should suffice. Please correct me if I'm wrong.

Reply:
Yes. One profile per engine and one engine per state, but profiles can't be not shared between engines. If you pass same pointer to profile pointer to each state, the pointer is shared between engines. It could be a problem for the cases that different engines/inferences are running in parallel.

Reply:
I got it. Thanks for explaining. I fixed this. Now we have one profile per state, and it's not tied to the provider.